WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift

```diff
@@ -218,7 +219,16 @@ extension RegisterDomainSuggestionsViewController: NUXButtonViewControllerDelega
             pushRegisterDomainDetailsViewController(domain)
         case .siteRedirect:
             setPrimaryButtonLoading(true)
-            createCartAndPresentWebView(domain)
+            createCart(domain) { [weak self] in
+                self?.presentWebViewForCurrentSite(domainSuggestion: domain)
+                self?.setPrimaryButtonLoading(false, afterDelay: 0.25)
+            }
+        case .mapped:
```

Comment:
Good point, I misunderstood `mapped` case.

I think then it's better to configure `RegisterDomainSuggestionViewController` not with `domainType` but by `flowType`, since we may want to have different configurations for same domain types.

Reply:
[I made a bunch of renamings](https://github.com/wordpress-mobile/WordPress-iOS/pull/20697/commits/d4c5fa41b927f8900edde861773511180daa223f) to reflect why `domainType` was used in these view controllers: to differentiate between different domain purchasing flows.

Hopefully, it will increase clarity. Clean-code-wise, it could probably be replaced by inheritance rather than `enum` to only have a couple of overridden methods that would differentiate between different cases. However, I don't think this usage of `enums` is too intrusive in this case.

Reply:
Great idea with the re-name, it looks good!